### ISSUE TITLE
Dataloader Prefetch data to GPU by cudaMemPrefetchAsync

### ISSUE THREAD

**Issue by Annie-Sihan-Chen** [pytorch/pytorch#50831]
There is a way to prefetch data between cpu and gpu by [cudaMemAdvise and  cudaMemPrefetchAsync](https://developer.nvidia.com/blog/beyond-gpu-memory-limits-unified-memory-pascal/). I am wondering that is this has been intergrated in to dataloader. I found a prefetch_factor, not sure if it is the one. If not, how can I integrated it?

**Comment by ngimel** [comment]
No, these functions have not been integrated. `prefetch_factor` means copying data ahead of time to the gpu memory with cudaMemcpyAsync.
